src/zenml/utils/env_utils.py

```diff
@@ -99,3 +108,36 @@ def reconstruct_environment_variables(
         # Remove the chunk environment variables
         for key in chunk_keys:
             env.pop(key)
+
+
+def substitute_env_variable_placeholders(value: V) -> V:
+    """Substitute environment variable placeholders in an object.
+
+    Args:
+        value: The object in which to substitute the placeholders.
+
+    Returns:
+        The object with placeholders substituted.
+    """
+
+    def _replace_with_env_variable_value(match: Match[str]) -> str:
+        key = match.group(1)
+        if value := os.getenv(key):
+            return value
+        else:
+            logger.debug(
+                "Unable to substitute environment variable placeholder %s "
+                "because the environment variable is not set, using an empty "
+                "string instead.",
+                key,
+            )
```

Comment:
Hm, here I have doubts... I know that Unix systems operate like that, but I don't like that, take `rm -rf $I_DONT_EXISTS` as an example - bad things might happen :)
Maybe raising and letting the user sort this out is safer?

Reply:
Don't have a strong opinion on this personally, but in most cases this will be harmless and worst case lead to a failed pipeline run, correct?

Reply:
Yes, but it will be harder to debug at the end, IMO. I will give you a green light here, but I would love to see an explicit raise here 🙂

Reply:
Raised a `KeyError` now but made it configurable on the util function at least in case we decide to change it at some point.